### ISSUE TITLE
Vulnerable to hashdos attack

### ISSUE THREAD

**Issue by LeSuisse** [watmough/jwHash#1]
This implementation is vulnerable to hashdos attack.
It is, indeed, quite easy to generate colliding strings with DJBX33A. You should use a keyed hash function like SipHash for exemple.

**Comment by rurban** [comment]
No. If you rely on a hash function only for hashdos attacks you are lost. Either use universal hashing,   a random seed or not a linked list. 

People usually use open addressing as it is faster and not vulnerable with 2 hash functions.

**Comment by LeSuisse** [comment]
I don't understand why you disagree with me.

I suggested the use of a PRF/keyed hash function like [SipHash](https://131002.net/siphash/) because it is the simple way to solve the issue. By the way, this is the solution adopted by Perl in 2003.

Others solutions involve bigger changes and are not easy to get it right.
